streams/src/main/java/org/apache/kafka/streams/kstream/internals/ProcessDetails.java

```diff
@@ -0,0 +1,341 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.streams.Consumed;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.errors.TopologyException;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.Produced;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.processor.StateStoreSupplier;
+import org.apache.kafka.streams.processor.StreamPartitioner;
+import org.apache.kafka.streams.processor.TimestampExtractor;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * Class used to hold the implementation details when making calls to inheritors of
+ * {@link AbstractStream} when building a topology
+ */
+@SuppressWarnings({"deprecation", "unchecked"})
+class ProcessDetails<K, V, S extends StateStore> {
+
+    private final Collection<String> sourceTopics;
+    private final String sinkTopic;
+    private final Pattern sourcePattern;
+    private final Produced<K, V> produced;
+    private final Consumed<K, V> consumed;
+    private final Materialized<K, V, S> materialized;
+    private final ProcessorSupplier<K, V> processorSupplier;
+    private MaterializedInternal<K, V, S> materializedInternal;
+    private ProducedInternal<K, V> producedInternal;
+    private ConsumedInternal<K, V> consumedInternal;
+    private final StateStoreSupplier<KeyValueStore<K, V>> storeSupplier;
+    private final StoreBuilder<KeyValueStore<K, V>> storeBuilder;
+    private final KTableSource<K, V> kTableSource;
+    private final String[] storeNames;
+    private final String connectProcessorName;
+    private final String[] parentNames;
+    private final String sourceName;
+
+
+    private ProcessDetails(final Collection<String> sourceTopics,
+                           final String sinkTopic,
+                           final Pattern sourcePattern,
+                           final Produced<K, V> produced,
+                           final Consumed<K, V> consumed,
+                           final Materialized<K, V, S> materialized,
+                           final ProcessorSupplier<K, V> processorSupplier,
+                           final StateStoreSupplier<KeyValueStore<K, V>> storeSupplier,
```

Comment:
I'd suggest we do not include any deprecated classes, because they may soon be removed in the next release.

Reply:
Ack.